data/test_multiline_intent_examples_yaml/nlu.yml

```diff
@@ -0,0 +1,14 @@
+version: "2.0"
+nlu:
+- intent: greet
+  examples:
+  - text: |
+      Hello,
+      My name is Christof
+  - text: |
+      Hello,
+      You can call me Christof
```

Comment:
how about also having a test for an example which is something like `- Hello,\nMy name is Christof`?

Reply:
TLDR: I think we should discourage using the \n notation in the training file and instead have people use literal line breaks if they want them. Having a uniform notation will lead to less unexpected automatic changes of the file structure. We could potentially add a warning if people use examples with escape chars

Long answer: This is a good question and relates to my thoughts I noted under *further work*. If you would read in a file containing - Hello,\nMy name is Christof? What you would actually be reading in is `\\n` which is an escaped backslash plus n. A real line break would just be a line break in the file. We could decode `\\n` to turn into `\n`, but then when it's rewritten, there isn't a simple way to know which notation was used originally, so we would probably not encode it and instead write a literal line break, thus leading to a change in the file. Even worse, an example might use both literal line breaks and `\n`, so you would need to track their positions if we wanted to make sure there are no changes, which seems way out of proportion to me.

Reply:
I personally agree that we should set some expectations and not try to support every possible way that people could write YAML. I guess that is pretty much the discussion in the "Revisit training data + IVC" ice box item.

Reply:
Sounds good. Should we create an issue to raise a warning for the `\n` case? 🤔